### DISCUSSION TITLE
Welcome to Bid For Power source code recreation!

### DISCUSSION THREAD

**Discussion by LegendaryGuard** [LegendaryGuard/BFP#1] [General]
### Please read this before posting:

The repository is about recreating Bid For Power source code, trying to be the same like BFP 1.2 version with the last compiled QVMs from 08/04/2002 and **must** be Quake 3 compatible (no extra PMF flags, modifying and adding any extra bit flags that would need to manipulate the engine code. So, that becomes incompatible to Quake 3 nature).
BFP never used engine code, only game code.

If you want to contribute, feel free to open a Pull Request (from your fork) and we can review the code. Please, read the [**contributing guide**](https://github.com/LegendaryGuard/BFP/discussions/7).

You can take a look in the [**documentations**](https://github.com/LegendaryGuard/BFP/tree/main/docs).

# Rules

Important, please read [**here**](https://github.com/LegendaryGuard/BFP/discussions/6).

# Priorities

- **1st and highest priority**: About implementing ki attack skills, BFP was largely based on Quake 3's weapon code, that part can be complex, so we didn't start from this because **principally**, we're trying to setup player statuses, polishing and adjusting animations, hit stun, aura, options, cvars, adjusting speed, camera, UI menu, HUD, melee, block, teleportation, powerlevel, ki energy counter adjustment, ... 

- **2nd priority**: When the 1st priority is done, we can implement gametypes: *Survival* (`g_gametype 3`), *Giant Monkey* (`g_gametype 4`) and *Last Man Standing* (`g_gametype 6`).

- **Last priority**: And after 2nd priority, we can implement the ki attacks and skin config file to set the explosion effects to setup as "[Custom plugin models](https://github.com/LegendaryGuard/BFP/blob/main/docs/Create_Custom_Models.md)" doc says. 

But it can use branches to test the stuff if you skip the priorities, but we can't accept if it isn't doing in this level order.
So, this way is less messy to develop and reach the goals. But it can use branches to test the stuff if you skip the priorities, but we can't accept if it is using the 2nd or last priority.

# Things to consider

The purpose on this repository isn't about adding DBZ and any other external features in the code, that could be developed if anyone wants to fork, not recommended though, because this is still incomplete and we don't give support and respond to complaints if anyone touched their code on its own.

We cannot accept any DBZ media, models and original art due to the copyright issues and GPL compliance, but anything related from BFP, yes (but not the models, maps and assets, such as the original contents without the author's permission). 
Just mannequin model and asset sources without copyrights and GPL compatible and some map sample without including any external copyrighted elements, to be added in [BFP-mediasource](https://github.com/LegendaryGuard/BFP-mediasource) repository.

**Comment by Lucasvilla98** [comment]
Hi! amazing proyect! i am making a mod of bid for power called ultra bid for power. since i cannot modify the ingame code i remade the entire UI and my mod works from the ui. if you can finish the cgame and game source code would be great! 
i have a backup of source code original menu of bfp remade by me. 

https://www.mediafire.com/file/chhp8ma76pohdwz/bfp-original.rar/file

it need some tweaks like color change for crosshair, unpack music option and a correct filter server code because i filtered it by game name.

it can be compiled with cygwin.
that code was made with ioquake3 as base

**Reply by Lucasvilla98** [reply]
i you have any form to write a private message give it to me :)

**Reply by LegendaryGuard** [reply]
> Hi! amazing proyect! i am making a mod of bid for power called ultra bid for power. since i cannot modify the ingame code i remade the entire UI and my mod works from the ui. if you can finish the cgame and game source code would be great! i have a backup of source code original menu of bfp remade by me.
> 
> https://www.mediafire.com/file/chhp8ma76pohdwz/bfp-original.rar/file
> 
> it need some tweaks like color change for crosshair, unpack music option and a correct filter server code because i filtered it by game name.
> 
> it can be compiled with cygwin. that code was made with ioquake3 as base

Hi, @Lucasvilla98 
I forgot to update the post in this discussion, so you can read what we were doing: https://github.com/LegendaryGuard/BFP/discussions/1#discussion-4666945

Great job with implementing the UI! But it doesn't match 100% to BFP.

Please, feel free to open a PR (Pull Request) about your contributions, so we can review the code and update.

> if you can finish the cgame and game source code would be great!

It's okay to cheer up, although this comment may sound asking too much of anyone, although it has been heard in the most places where some project is being started and/or enhanced. lol
This development project shouldn't be involved by a single person, it's among everyone to achieve these main goals.
I was trying to maintain the aura vectors to be the same as the player positioning under legs, also to make it rotate horizontally (the yaw); and developing the hit stun timer, we haven't managed to complete that though.

Also feel free to contribute. :)
